src/nnet3/nnet-optimize.cc

```diff
@@ -432,6 +433,134 @@ void CachingOptimizingCompiler::UpdateCache(const ComputationRequest *request,
                             std::make_pair(computation, ait)));
 }
 
+void CachingOptimizingCompiler::ReadCache(std::istream &is, bool binary) {
+  // we won't read cached computations if any optimize option has been changed.
+  bool read_cache = true;
+  ExpectToken(is, binary, "<NnetOptimizeOptions>");
```

Comment:
It would be more straightforward to give a Read and Write function and an == operator to the options class.